agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -1427,3 +1427,45 @@ func TestElasticInferenceValidator(t *testing.T) {
 	exitCode, _ := task.ContainerExitcode("container_1")
 	assert.Equal(t, 42, exitCode, fmt.Sprintf("Expected exit code of 42 for container; got %d", exitCode))
 }
+
+// TestServerEndpointValidator tests the workflow of task server endpoint
+func TestServerEndpointValidator(t *testing.T) {
```

Comment:
Is there a particular reason you hand-wrote this code instead of using the simpletests package?  I would recommend using that instead.

Reply:
I believe it is for setting the task iam roles configuration.

Reply:
Discussed with Sam offline, we might not need network mode setup for it to work. 

But I'm thinking for now, let's keep test file as hand-written to release it today to solve our customers' issue. I will keep troubleshooting to see if I can make functional test work with simpletests file

Reply:
Task IAM roles are supported with "host", "bridge", and "awsvpc" network modes, and we have existing functional tests that validate this.  There should not be a dependency on using the "host" network mode.

Reply:
Read some document, agree on this should not depend on host mode. I will dive deep to see if my host needs some config to make it functional test on other modes, and I will follow up with another pr to address it once i get it figured out.